АНАЛИЗ списка спасенных людей с затонувшего корабля Titanic

In [1]:
# Библиотека для работы с табличными данными
import pandas as pd

In [2]:
ds = pd.read_csv('sample_data/titanic.csv')
ds

Survived  Pclass                                               Name  \
0           0       3                             Mr. Owen Harris Braund   
1           1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2           1       3                              Miss. Laina Heikkinen   
3           1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4           0       3                            Mr. William Henry Allen   
..        ...     ...                                                ...   
882         0       2                               Rev. Juozas Montvila   
883         1       1                        Miss. Margaret Edith Graham   
884         0       3                     Miss. Catherine Helen Johnston   
885         1       1                               Mr. Karl Howell Behr   
886         0       3                                 Mr. Patrick Dooley   

        Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0      male  22.0                        1                        0   7.2500  
1    female  38.0                        1                        0  71.2833  
2    female  26.0                        0                        0   7.9250  
3    female  35.0                        1                        0  53.1000  
4      male  35.0                        0                        0   8.0500  
..      ...   ...                      ...                      ...      ...  
882    male  27.0                        0                        0  13.0000  
883  female  19.0                        0                        0  30.0000  
884  female   7.0                        1                        2  23.4500  
885    male  26.0                        0                        0  30.0000  
886    male  32.0                        0                        0   7.7500  

[887 rows x 8 columns]

In [4]:
ds.describe()

Survived      Pclass         Age  Siblings/Spouses Aboard  \
count  887.000000  887.000000  887.000000               887.000000   
mean     0.385569    2.305524   29.471443                 0.525366   
std      0.487004    0.836662   14.121908                 1.104669   
min      0.000000    1.000000    0.420000                 0.000000   
25%      0.000000    2.000000   20.250000                 0.000000   
50%      0.000000    3.000000   28.000000                 0.000000   
75%      1.000000    3.000000   38.000000                 1.000000   
max      1.000000    3.000000   80.000000                 8.000000   

       Parents/Children Aboard       Fare  
count               887.000000  887.00000  
mean                  0.383315   32.30542  
std                   0.807466   49.78204  
min                   0.000000    0.00000  
25%                   0.000000    7.92500  
50%                   0.000000   14.45420  
75%                   0.000000   31.13750  
max                   6.000000  512.32920

В данном случае у нас следующие строки:



1.   Survived - Количество выживших
2.   Pclass - кто-бы знал
3.   Age - Возраст
4.   Siblings/Spouses Aboard - Детей/Супругов, оставшихся на борту
5.   Parents/Children Aboard - Родителей/детей, оставшихся на борту
6.   Fare - Транспортные расходы

In [11]:
ds[(ds['Age'] < 1) & (ds['Parents/Children Aboard'] > 1)]

Survived  Pclass                           Name   Sex   Age  \
77          1       2   Master. Alden Gates Caldwell  male  0.83   
303         1       1  Master. Hudson Trevor Allison  male  0.92   

     Siblings/Spouses Aboard  Parents/Children Aboard    Fare  
77                         0                        2   29.00  
303                        1                        2  151.55

РЕЗУЛЬТАТ ПОИСКА Из 887 спасенных людей всего детей до 1 года потерявших обоих родителей 2.

In [16]:
ds[(ds['Age'] > 75)]

Survived  Pclass                                 Name   Sex   Age  \
627         1       1  Mr. Algernon Henry Wilson Barkworth  male  80.0   

     Siblings/Spouses Aboard  Parents/Children Aboard  Fare  
627                        0                        0  30.0

РЕЗУЛЬТАТ ПОИСКА Самому старому спасенному человеку 80 лет

In [17]:
print(ds['Age'].sum())
# Общий возраст всех спасенных

26141.17


In [24]:
print(ds['Age'].mean())
# Средний возраст всех спасенных

29.471443066516347


РЕЗУЛЬТАТ ПОИСКА Средний возраст спасенных около 30 лет. Общий вес более 26 т